File: C++-Problems/Archived Problems - Project Euler/problem3.cpp
#include <iostream>
using namespace std;
void FindLargestPrimeFactor(long long &Number, long long &LargestPrimeFactor, long long &CurrentFactor)
{
	Number = 600851475143;
	LargestPrimeFactor = 1;
	CurrentFactor = 2;
	while (Number > 1)
	{
		if (Number % CurrentFactor == 0)
		{
			LargestPrimeFactor = CurrentFactor;
			Number /= CurrentFactor;
		}
		else
		{
			CurrentFactor++;
		}
	}
	cout << "The largest prime factor of 600851475143 is: " << LargestPrimeFactor << endl;
}
int main()
{
	long long  Number, LargestPrimeFactor, CurrentFactor;
	FindLargestPrimeFactor(Number, LargestPrimeFactor, CurrentFactor);
	return 0;
}

File: JavaScript-Problems/Archived Problems - Project Euler/problem3.js
/* 
* Finds the largest prime factor of the number 600851475143.
 *
 * It uses the BigInt type for accurate handling of large numbers.
 * The algorithm iteratively divides the number by the smallest possible factor
 * until the number is reduced to 1. The last factor successfully used is the largest prime factor.
 *
*/

function FindLargestPrimeFactor(){
    let number = 600851475143n; 
    let largestFactor = 1n;
    let currentFactor = 2n;

    const initialNumber = number;

    while( number > 1n){
        if ( number % currentFactor === 0n){
            largestFactor = currentFactor;
            number /= currentFactor;
        }
        else{
            currentFactor++;
        }
    }
    console.log(`The Largest Prime Factor of ${initialNumber} is: ${largestFactor.toString()}`);
    return largestFactor;
}

FindLargestPrimeFactor();